include/h2o.h

```diff
@@ -2038,10 +2047,18 @@ void h2o_status_register_configurator(h2o_globalconf_t *conf);
 
 /* lib/handler/headers_util.c */
 
+struct st_yoml_t;
+struct st_h2o_headers_add_arg_t {
+    struct st_yoml_t *node;
+    h2o_iovec_t *name;
+    h2o_iovec_t value;
+};
+typedef struct st_h2o_headers_add_arg_t h2o_headers_add_arg_t;
+
 /**
  * appends a headers command to the list
  */
-void h2o_headers_append_command(h2o_headers_command_t **cmds, int cmd, h2o_iovec_t *name, h2o_iovec_t value,
+void h2o_headers_append_command(h2o_headers_command_t **cmds, int cmd, h2o_headers_add_arg_t *args, size_t num_args,
```

Comment:
Due to the reason stated above, I think we might prefer using a typedef-ed version of `st_h2o_headers_command_arg_t` here.

Reply:
I think that `h2o_headers_append_command` must have access to the definition of the struct:
```
    for (i = 0; i < num_args; i++) {
        new_cmds[cnt].args[i].name = args[i].name;
        new_cmds[cnt].args[i].value = args[i].value;
    }  
```
We could have a struct that doesn't have the node in it, what do you think?

Reply:
> We could have a struct that doesn't have the node in it, what do you think?

Yes. This is what I meant (and that `st_h2o_headers_command_arg_t` meets that requirement once it gets a typedef'ed name).

Reply:
I'm not 100% sure i follow. I've tried to do so in 17df4791b, but we still need the definition to be in a header because it's used by both `lib/handler/headers_util.c` and `lib/handler/configurator/headers_util.c`, am i missing something?

Reply:
Discussed offline; went with c08ed1b.